### PR TITLE
Add support for gzip compression

### LINKED ISSUE #39: Add support for gzip response Content-Encoding
ClickHouse allows for compression on the HTTP interface
https://clickhouse.yandex/docs/en/interfaces/http_interface/

> It is […] possible to use the standard gzip-based HTTP compression. To send a POST request compressed using gzip, append the request header Content-Encoding: gzip. In order for ClickHouse to compress the response using gzip, you must append Accept-Encoding: gzip to the request headers[…]

The driver should support that.

### PR DESCRIPTION
**Pull request by maciej**
This PR introduces gzip response compression. It's a Work-In-Progress, tests are missing. It's based off #38.

Resolves #39

### CODE REVIEW

**Comment by maciej** [comment]
The PR is now rebased against recent master. I've included the current `conn_test.go` query tests to run on a HTTP connection that requests compressed responses.
Tests pass both locally and on Travis.
@bgaifullin I'd appreciate if you could take the time to review it!

File: conn.go [hunk 1]
```diff
@@ -210,7 +213,15 @@ func (c *conn) doRequest(ctx context.Context, req *http.Request) (io.ReadCloser,
 		return nil, err
 	}
 
-	return resp.Body, nil
+	respReadCloser := resp.Body
```
[the remark this answers is not in the document]
Comment (maciej) [reply conn.go]: Done

File: conn_test.go [hunk 2]
```diff
@@ -44,6 +44,7 @@ func (s *connSuite) TestQuery() {
 		},
 	}
 
+	// Tests on regular connection
 	for _, tc := range testCases {
 		rows, err := s.conn.Query(tc.query, tc.args...)
```
[the remark this answers is not in the document]
Comment (maciej) [reply conn_test.go]: done

**Comment by maciej** [comment]
@bgaifullin I've added both of your comments and pushed an amended version.

**Comment by bgaifullin** [comment]
LGTM. thanks.

**Comment by maciej** [comment]
:+1:

**Pull request merged**